_episodes/04-sharing-environments.md

```diff
@@ -349,63 +354,77 @@ The `--prune` option tells Conda to remove any dependencies that are no longer r
 > {: .solution}
 {: .challenge}
 
-> ## Installing via `pip` in `environment.yml` files
->
-> Since you write `environment.yml` files for all of your projects, you might be wondering how
-> to specify that packages should be installed using `pip` in the `environment.yml` file.  Here
-> is an example `environment.yml` file that uses `pip` to install the `kaggle` and `yellowbrick`
-> packages.
->
-> ~~~
-> name: example
+## Installing via `pip` in `environment.yml` files
+
+Since you write `environment.yml` files for all of your projects, you might be wondering how to specify that packages
+should be installed using `pip` in the `environment.yml` file.  Here is an example `environment.yml` file that uses
+`pip` to install the `kaggle` and `yellowbrick` packages.
+
+~~~
+name: example
+
+dependencies:
+  - jupyterlab=1.0
+  - matplotlib=3.1
+  - pandas=0.24
+  - scikit-learn=0.21
+  - pip=22.3
+  - pip:
+    - kaggle
+    - yellowbrick==1.5
+~~~
+
+> Note two things...
 >
-> dependencies:
->  - jupyterlab=1.0
->  - matplotlib=3.1
->  - pandas=0.24
->  - scikit-learn=0.21
->  - pip=22.3
->  - pip:
->    - kaggle==1.5
->    - yellowbrick==1.5
-> ~~~
+> 1. `pip` is installed as a dependency under conda **first** (under the `dependencies` section) with an explicit

```

Comment:
Better to install Python as pip is included?

Reply:
My notes here are because I wanted to explain the nesting levels in the YAML file. when the environment is exported Python (i.e. `conda env export --from-history`) then even though the environment was created explicitly with `python=3.10` (if everything is now consistent!) Python _isn't_ listed in the exports, it appears to be inferred from the `dependencies:` list.

`pip` isn't listed there either, even though installing `python` in an environment automatically pulls in `pip`.

In this section we are explaining how to include packages that were installed via `pip` and until the issue you found addresses this we have to advise that its done by editing the YAML file.

As Python will be installed as a dependency of all Python packages (including `pip`) I think its simplest to have `pip[=22.3]` (i.e. optional version) listed under `dependencies:` and then list the packages it should install.

Reply:
That's more nuanced than I'd thought. Makes sense though!